pkg/kubelet/config/config.go

```diff
@@ -78,20 +73,15 @@ func NewPodConfig(mode PodConfigNotificationMode) *PodConfig {
 // Channel creates or returns a config source channel.  The channel
 // only accepts PodUpdates
 func (c *PodConfig) Channel(source string) chan<- interface{} {
-	c.sourcesLock.Lock()
-	defer c.sourcesLock.Unlock()
-	c.sources.Insert(source)
 	return c.mux.Channel(source)
 }
 
-// SeenAllSources returns true if this config has received a SET
-// message from all configured sources, false otherwise.
-func (c *PodConfig) SeenAllSources() bool {
+func (c *PodConfig) SourceSeen(source string) bool {
 	if c.pods == nil {
 		return false
 	}
-	glog.V(6).Infof("Looking for %v, have seen %v", c.sources.List(), c.pods.sourcesSeen)
-	return c.pods.seenSources(c.sources.List()...)
+	glog.V(6).Infof("Looking for %v, have seen %v", source, c.pods.sourcesSeen)
```

Comment:
If I understand, this function is testing whether a source has been seen, not asserting that it has.  Maybe rename to IsSourceSeen()